docs/tutorials/custom_segmentation_trainer.ipynb

```diff
@@ -0,0 +1,514 @@
+    "            ModelCheckpoint(every_n_epochs=50, save_top_k=-1, save_last=True),\n",
+    "            ModelCheckpoint(monitor=self.monitor, mode=self.mode, save_top_k=5),\n",
+    "        ]\n",
+    "\n",
+    "    def on_train_epoch_start(self) -> None:\n",
+    "        \"\"\"Log the learning rate at the start of each training epoch.\"\"\"\n",
+    "        optimizers = self.optimizers()\n",
+    "        if isinstance(optimizers, list):\n",
+    "            lr = optimizers[0].param_groups[0]['lr']\n",
+    "        else:\n",
+    "            lr = optimizers.param_groups[0]['lr']\n",
+    "        self.logger.experiment.add_scalar('lr', lr, self.current_epoch)  # type: ignore"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Train model\n",
+    "\n",
+    "The remainder of the turial is straightforward and follows the typical [PyTorch Lightning](https://lightning.ai/) training routine. We instantiate a `DataModule` for the LandCover.AI 100 dataset (a small version of the LandCover.AI dataset for notebook testing), instantiate a `CustomSemanticSegmentationTask` with a U-Net and ResNet-50 backbone, then train the model using a Lightning trainer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dm = LandCoverAI100DataModule(root='data', batch_size=10, num_workers=2, download=True)\n",
+    "\n",
+    "# You can use the following for actual training runs\n",
+    "# from torchgeo.datamodules import LandCoverAIDataModule\n",
+    "# dm = LandCoverAIDataModule(root='data/', batch_size=64, num_workers=8, download=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "task = CustomSemanticSegmentationTask(\n",
+    "    model='unet',\n",
+    "    backbone='resnet50',\n",
+    "    weights=True,\n",
+    "    in_channels=3,\n",
+    "    num_classes=6,\n",
+    "    loss='ce',\n",
+    "    lr=1e-3,\n",
+    "    tmax=50,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "\"backbone\":        resnet50\n",
+       "\"class_weights\":   None\n",
+       "\"eta_min\":         1e-06\n",
+       "\"freeze_backbone\": False\n",
+       "\"freeze_decoder\":  False\n",
+       "\"ignore_index\":    None\n",
+       "\"in_channels\":     3\n",
+       "\"loss\":            ce\n",
+       "\"lr\":              0.001\n",
+       "\"model\":           unet\n",
+       "\"num_classes\":     6\n",
+       "\"num_filters\":     3\n",
+       "\"patience\":        10\n",
+       "\"tmax\":            50"
+      ]
+     },
+     "execution_count": null,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# validate that the task's hyperparameters are as expected\n",
```

Comment:
At some point we transitioned from markdown to code comments. I would prefer markdown.